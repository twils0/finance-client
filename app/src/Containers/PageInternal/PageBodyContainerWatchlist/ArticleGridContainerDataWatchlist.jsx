import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { statusNames } from '../../../Constants/dataConstantsWatchlist';

import ArticleGridDataWatchlist from '../../../Components/PageInternal/PageBodyWatchlist/Article/ArticleGridDataWatchlist';

const ArticleGridContainerDataWatchlist = (props) => {
  const {
    securities, articleGridWidthIndent, articleGridHeightBuffer, articleGridHeightIndent,
  } = props;

  const { current } = securities;

  if (current && securities[current]) {
    const currentSecurity = securities[current];

    if (currentSecurity.data && currentSecurity.data.price) {
      const dataPrice = currentSecurity.data.price;

      return (
        <ArticleGridDataWatchlist
          widthIndent={articleGridWidthIndent}
          heightBuffer={articleGridHeightBuffer}
          heightIndent={articleGridHeightIndent}
          dataPrice={dataPrice}
        />
      );
    }
  }
  return null;
};

ArticleGridContainerDataWatchlist.propTypes = {
  articleGridWidthIndent: PropTypes.number.isRequired,
  articleGridHeightBuffer: PropTypes.number.isRequired,
  articleGridHeightIndent: PropTypes.number.isRequired,
  securities: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  articleGridWidthIndent: state.ui.internal.watchlist.utilities.articleGrid.widthIndent,
  articleGridHeightBuffer: state.ui.internal.watchlist.utilities.articleGrid.heightBuffer,
  articleGridHeightIndent: state.ui.internal.watchlist.utilities.articleGrid.heightIndent,
  statusSecurityData: state.data.watchlist.status[statusNames.GET_SECURITY_DATA].status,
  securities: state.data.watchlist.securities,
});

export default connect(mapStateToProps)(ArticleGridContainerDataWatchlist);
