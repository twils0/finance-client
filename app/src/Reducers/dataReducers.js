import { combineReducers } from 'redux';

import auth from './dataReducersAuth';
import aws from './dataReducersAWS';
import watchlist from './dataReducersWatchlist';
import account from './dataReducersAccount';

const dataReducers = combineReducers({
  auth,
  aws,
  watchlist,
  account,
});

export default dataReducers;
