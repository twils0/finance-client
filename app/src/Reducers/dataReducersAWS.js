import { combineReducers } from 'redux';

import { requestStatusTypes } from '../Constants/universalConstants';
import { actionTypes } from '../Constants/dataConstantsAWS';

export const initialState = {
  status: requestStatusTypes.IDLE,
  user: null,
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_AWS_STATUS:
      return action.payload.status;
    case actionTypes.RESET_AWS_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const user = (state = initialState.user, action) => {
  switch (action.type) {
    case actionTypes.SET_AWS_USER:
      return action.payload.user;
    case actionTypes.RESET_AWS_STATE:
      return initialState.user;
    default:
      return state;
  }
};

const dataReducersAWS = combineReducers({
  status,
  user,
});

export default dataReducersAWS;
