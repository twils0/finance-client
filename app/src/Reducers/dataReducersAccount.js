import { combineReducers } from 'redux';

import { requestStatusTypes } from '../Constants/universalConstants';
import { actionTypes, statusNames, fieldNames } from '../Constants/dataConstantsAccount';

export const initialState = {
  status: {
    [statusNames.AWS_FIELDS]: {
      id: statusNames.AWS_FIELDS,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.STRIPE_FIELDS]: {
      id: statusNames.STRIPE_FIELDS,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_STRIPE_FIELDS_TOKEN]: {
      id: statusNames.UPDATE_STRIPE_FIELDS_TOKEN,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_STRIPE_FIELDS_REQUEST]: {
      id: statusNames.UPDATE_STRIPE_FIELDS_REQUEST,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_DB_FIELDS]: {
      id: statusNames.UPDATE_DB_FIELDS,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_AWS_FIELDS]: {
      id: statusNames.UPDATE_AWS_FIELDS,
      status: requestStatusTypes.IDLE,
    },
  },
  fields: {
    list: [...Object.values(fieldNames)],
    [fieldNames.NAME]: {
      id: fieldNames.NAME,
      value: '',
    },
    [fieldNames.EMAIL]: {
      id: fieldNames.EMAIL,
      value: '',
    },
    [fieldNames.EMAIL_ADDITIONAL]: {
      id: fieldNames.EMAIL_ADDITIONAL,
      value: '',
    },
    [fieldNames.PHONE]: {
      id: fieldNames.PHONE,
      value: '',
    },
    [fieldNames.NAME_ON_CARD]: {
      id: fieldNames.NAME_ON_CARD,
      value: '',
    },
    [fieldNames.PROMO_CODE]: {
      id: fieldNames.PROMO_CODE,
      value: '',
    },
    [fieldNames.PROMO_CODE_VALID]: {
      id: fieldNames.PROMO_CODE_VALID,
      value: true,
    },
    [fieldNames.CUSTOMER]: {
      id: fieldNames.CUSTOMER,
      value: '',
    },
  },
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_ACCOUNT_STATUS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_ACCOUNT_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const fields = (state = initialState.fields, action) => {
  switch (action.type) {
    case actionTypes.SET_FIELD:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.SET_FIELDS:
      return {
        ...state,
        ...action.payload,
      };
    case actionTypes.RESET_ACCOUNT_STATE:
      return initialState.fields;
    default:
      return state;
  }
};

const dataReducersAccount = combineReducers({
  status,
  fields,
});

export default dataReducersAccount;
