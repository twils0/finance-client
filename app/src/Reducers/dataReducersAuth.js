import { combineReducers } from 'redux';

import { requestStatusTypes, pathNames } from '../Constants/universalConstants';
import { actionTypes, statusNames, codeTypeNames } from '../Constants/dataConstantsAuth';

export const initialState = {
  authenticated: false,
  redirectURL: pathNames.WATCHLIST,
  status: {
    [statusNames.LOGIN]: {
      id: statusNames.LOGIN,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.SIGN_OUT_DEVICES]: {
      id: statusNames.SIGN_OUT_DEVICES,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.LOGIN_MFA]: {
      id: statusNames.LOGIN_MFA,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.SIGN_UP]: {
      id: statusNames.SIGN_UP,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_PHONE]: {
      id: statusNames.VERIFY_PHONE,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_PHONE_CODE]: {
      id: statusNames.VERIFY_PHONE_CODE,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_EMAIL]: {
      id: statusNames.VERIFY_EMAIL,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_EMAIL_LINK]: {
      id: statusNames.VERIFY_EMAIL_LINK,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.FORGOT_PASSWORD]: {
      id: statusNames.FORGOT_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.RESET_PASSWORD]: {
      id: statusNames.RESET_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.CHANGE_PASSWORD]: {
      id: statusNames.CHANGE_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.LOGOUT]: {
      id: statusNames.LOGOUT,
      status: requestStatusTypes.SUCCESS,
    },
    [statusNames.DELETE_ACCOUNT]: {
      id: statusNames.DELETE_ACCOUNT,
      status: requestStatusTypes.SUCCESS,
    },
  },
  codeTypes: {
    [codeTypeNames.VERIFY_PHONE]: {
      id: codeTypeNames.VERIFY_PHONE,
      needed: true,
    },
    [codeTypeNames.VERIFY_EMAIL]: {
      id: codeTypeNames.VERIFY_EMAIL,
      needed: true,
    },
    [codeTypeNames.VERIFY_EMAIL_ADDITIONAL]: {
      id: codeTypeNames.VERIFY_EMAIL_ADDITIONAL,
      needed: true,
    },
  },
};

export const authenticated = (state = initialState.authenticated, action) => {
  switch (action.type) {
    case actionTypes.SET_AUTHENTICATED:
      return action.payload.authenticated;
    case actionTypes.RESET_AUTH_STATE:
      return initialState.authenticated;
    default:
      return state;
  }
};

export const redirectURL = (state = initialState.redirectURL, action) => {
  switch (action.type) {
    case actionTypes.SET_REDIRECT_URL:
      return action.payload.redirectURL;
    default:
      return state;
  }
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_AUTH_STATUS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_AUTH_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const codeTypes = (state = initialState.codeTypes, action) => {
  switch (action.type) {
    case actionTypes.SET_CODE_TYPE:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_AUTH_STATE:
      return initialState.codeTypes;
    default:
      return state;
  }
};

const dataReducersAuth = combineReducers({
  authenticated,
  redirectURL,
  status,
  codeTypes,
});

export default dataReducersAuth;
