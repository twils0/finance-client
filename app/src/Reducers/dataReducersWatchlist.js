import { combineReducers } from 'redux';

import { requestStatusTypes } from '../Constants/universalConstants';
import { actionTypes, statusNames } from '../Constants/dataConstantsWatchlist';

export const initialState = {
  status: {
    [statusNames.SECURITIES]: {
      id: statusNames.SECURITIES,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.GET_SECURITY_DATA]: {
      id: statusNames.GET_SECURITY_DATA,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_SECURITIES]: {
      id: statusNames.UPDATE_SECURITIES,
      status: requestStatusTypes.IDLE,
    },
  },
  securities: {
    current: null,
    list: [],
  },
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_WATCHLIST_STATUS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_WATCHLIST_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const securities = (state = initialState.securities, action) => {
  switch (action.type) {
    case actionTypes.SET_SECURITIES_CURRENT:
      return {
        ...state,
        ...action.payload,
      };
    case actionTypes.SET_SECURITIES_LIST:
      return {
        ...state,
        ...action.payload,
      };
    case actionTypes.SET_SECURITIES_ALL:
      return {
        ...action.payload.securities,
      };
    case actionTypes.SET_SECURITY:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.SET_SECURITY_DATA: {
      if (!state[action.payload.id]) {
        return {
          ...state,
          [action.payload.id]: {
            ...state[action.payload.id],
            ...action.payload,
          },
        };
      }

      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          data: {
            ...state[action.payload.id].data,
            ...action.payload.data,
          },
        },
      };
    }
    case actionTypes.RESET_WATCHLIST_STATE:
      return initialState.securities;
    default:
      return state;
  }
};

const dataReducersWatchlist = combineReducers({
  status,
  securities,
});

export default dataReducersWatchlist;
