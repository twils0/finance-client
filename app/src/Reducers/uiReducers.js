import { combineReducers } from 'redux';

import app from './uiReducersApp';
import external from './uiReducersExternal';
import internal from './uiReducersInternal';

const uiReducers = combineReducers({
  app,
  external,
  internal,
});

export default uiReducers;
