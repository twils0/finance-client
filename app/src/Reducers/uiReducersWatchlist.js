import { combineReducers } from 'redux';

export const initialState = {
  utilities: {
    pageBody: {
      mediaWidth: 925,
    },
    list: {
      springSettings: { stiffness: 300, damping: 50 },
      width: 300,
      widthBufferLeft: 30,
      widthBufferRight: 15,
      widthBufferScrollBar: 20,
      heightBuffer: 60,
      rowWidthIndent: 10,
      rowHeight: 61,
      rowHeightIndent: 5,
      rowHeightBufferBottom: 10,
      burgerMenuIconWidth: 30,
      burgerMenuIconWidthIndent: 10,
      burgerMenuIconHeight: 27,
    },
    separatorLine: {
      width: 4,
      heightBuffer: 40,
    },
    article: {
      widthBuffer: 60,
      widthIndent: 30,
      mediaWidth: 700,
      heightBuffer: 60,
      heightIndent: 30,
    },
    articleMenu: {
      height: 75,
    },
    articleMenuSecurity: {
      widthBuffer: 40,
      heightBuffer: 20,
    },
    articleGrid: {
      widthBuffer: 20,
      widthIndent: 10,
      heightBuffer: 20,
      heightIndent: 10,
    },
    articleTable: {
      width: 470,
      widthBuffer: 20,
      heightBuffer: 20,
    },
    articleDescription: {
      minWidth: 250,
      widthBuffer: 20,
      heightBuffer: 20,
    },
  },
};

initialState.utilities.list.responsiveHeight = 4 * (initialState.utilities.list.rowHeight + initialState.utilities.list.rowHeightBufferBottom);

initialState.utilities.article.outsideWidthTotal = initialState.utilities.list.width
  + initialState.utilities.list.widthBufferLeft
  + initialState.utilities.list.widthBufferRight
  + initialState.utilities.list.widthBufferScrollBar
  + initialState.utilities.separatorLine.width
  + initialState.utilities.article.widthBuffer;

export const utilities = (state = initialState.utilities) => state;

const uiReducersWatchlist = combineReducers({
  utilities,
});

export default uiReducersWatchlist;
