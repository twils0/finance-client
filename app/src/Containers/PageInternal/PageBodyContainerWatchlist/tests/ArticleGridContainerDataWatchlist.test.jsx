import React from 'react';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import { initialState as initialStateWatchlist } from '../../../../Reducers/dataReducersWatchlist';
import { initialState as initialStateUIWatchlist } from '../../../../Reducers/uiReducersWatchlist';

import ArticleGridContainerDataWatchlist from '../ArticleGridContainerDataWatchlist';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

const shallowComponent = store => shallow(<ArticleGridContainerDataWatchlist />, {
  context: {
    store,
  },
}).dive();

const testId0 = 'testId0';
const testId1 = 'testId1';
const testId2 = 'testId2';

const securities = {
  current: testId0,
  list: [testId0, testId1, testId2],
  [testId0]: {
    id: testId0,
    name: 'testName0',
    category: 'testCategory0',
    exchange: 'testExchange0',
    tickerCusip: 'testTickerCusip0',
    data: {
      description: 'testDescription0',
    },
  },
  [testId1]: {
    id: testId1,
    name: 'testName1',
    category: 'testCategory1',
    exchange: 'testExchange1',
    tickerCusip: 'testTickerCusip1',
    data: {
      description: 'testDescription1',
    },
  },
  [testId2]: {
    id: testId2,
    name: 'testName2',
    category: 'testCategory2',
    exchange: 'testExchange2',
    tickerCusip: 'testTickerCusip2',
    data: {
      description: 'testDescription2',
    },
  },
};

describe('Containers', () => {
  describe('PageInternal', () => {
    describe('PageBodyContainerWatchlist', () => {
      describe('ArticleGridContainerDataWatchlist', () => {
        it('shallow renders correctly', async () => {
          const stateBeforeWatchlist = JSON.parse(JSON.stringify(initialStateWatchlist));

          stateBeforeWatchlist.securities = securities;

          const store = mockStore({
            data: {
              watchlist: stateBeforeWatchlist,
            },
            ui: {
              internal: {
                watchlist: initialStateUIWatchlist,
              },
            },
          });

          const expectedActions = [];

          const wrapper = shallowComponent(store);

          const actions = store.getActions();

          expect(actions).toEqual(expectedActions);
          expect(wrapper).toMatchSnapshot();
        });

        it('shallow renders correctly, with no security data', async () => {
          const stateBeforeWatchlist = JSON.parse(JSON.stringify(initialStateWatchlist));

          const noDataSecurities = JSON.parse(JSON.stringify({ ...securities }));
          delete noDataSecurities.testId0.data;

          stateBeforeWatchlist.securities = noDataSecurities;

          const store = mockStore({
            data: {
              watchlist: stateBeforeWatchlist,
            },
            ui: {
              internal: {
                watchlist: initialStateUIWatchlist,
              },
            },
          });

          const expectedActions = [];

          const wrapper = shallowComponent(store);

          const actions = store.getActions();

          expect(actions).toEqual(expectedActions);
          expect(wrapper).toMatchSnapshot();
        });
      });
    });
  });
});
