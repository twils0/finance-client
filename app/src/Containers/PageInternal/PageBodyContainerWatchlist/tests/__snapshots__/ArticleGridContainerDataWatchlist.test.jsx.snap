// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers PageInternal PageBodyContainerWatchlist ArticleGridContainerDataWatchlist shallow renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleGridContainerDataWatchlist
    articleGridHeightBuffer={20}
    articleGridHeightIndent={10}
    articleGridWidthIndent={10}
    dispatch={[Function]}
    items={
      Object {
        "current": "testId0",
        "list": Array [
          "testId0",
          "testId1",
          "testId2",
        ],
        "testId0": Object {
          "category": "testCategory0",
          "data": Object {
            "description": "testDescription0",
          },
          "exchange": "testExchange0",
          "id": "testId0",
          "name": "testName0",
          "tickerCusip": "testTickerCusip0",
        },
        "testId1": Object {
          "category": "testCategory1",
          "data": Object {
            "description": "testDescription1",
          },
          "exchange": "testExchange1",
          "id": "testId1",
          "name": "testName1",
          "tickerCusip": "testTickerCusip1",
        },
        "testId2": Object {
          "category": "testCategory2",
          "data": Object {
            "description": "testDescription2",
          },
          "exchange": "testExchange2",
          "id": "testId2",
          "name": "testName2",
          "tickerCusip": "testTickerCusip2",
        },
      }
    }
    statusItemData="IDLE"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "hydrateIn": undefined,
  },
}
`;

exports[`Containers PageInternal PageBodyContainerWatchlist ArticleGridContainerDataWatchlist shallow renders correctly, with no item data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleGridContainerDataWatchlist
    articleGridHeightBuffer={20}
    articleGridHeightIndent={10}
    articleGridWidthIndent={10}
    dispatch={[Function]}
    items={
      Object {
        "current": "testId0",
        "list": Array [
          "testId0",
          "testId1",
          "testId2",
        ],
        "testId0": Object {
          "category": "testCategory0",
          "exchange": "testExchange0",
          "id": "testId0",
          "name": "testName0",
          "tickerCusip": "testTickerCusip0",
        },
        "testId1": Object {
          "category": "testCategory1",
          "data": Object {
            "description": "testDescription1",
          },
          "exchange": "testExchange1",
          "id": "testId1",
          "name": "testName1",
          "tickerCusip": "testTickerCusip1",
        },
        "testId2": Object {
          "category": "testCategory2",
          "data": Object {
            "description": "testDescription2",
          },
          "exchange": "testExchange2",
          "id": "testId2",
          "name": "testName2",
          "tickerCusip": "testTickerCusip2",
        },
      }
    }
    statusItemData="IDLE"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "hydrateIn": undefined,
  },
}
`;
